pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -489,6 +477,29 @@ func (s *DiscoveryServer) WorkloadUpdate(id string, workloadLabels map[string]st
 
 	adsLog.Infof("Label change, full push %s ", id)
 	s.ConfigUpdate(&model.PushRequest{Full: true})
```

Comment:
I think we should simply not do this. Instead, tell people that they should never change pod labels after it has started. For sake of performance.

Reply:
I agree there is really a rare case changing labels. But as this is a feature lived long ago, it is not easy to deprecate this and eventually remove it.